doc/profili-di-interazione/profili-non-bloccanti.rst

```diff
@@ -0,0 +1,1448 @@
+|                                 |     {                                                                  |
+|                                 |       "c": "OK"                                                        |
+|                                 |     }                                                                  |
++---------------------------------+------------------------------------------------------------------------+
+| 4 Response Header & Body        | .. code-block:: JSON                                                   |
+| (HTTP Status Code 202 Accepted) |                                                                        |
+|                                 |     {                                                                  |
+|                                 |       "result" : "ACK"                                                 |
+|                                 |     }                                                                  |
++---------------------------------+------------------------------------------------------------------------+
+
+.. _interfaccia-soap-1:
+
+Interfaccia SOAP
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Nel caso di implementazione mediante tecnologia SOAP, l’endpoint di
+callback ed il correlation ID, vengono inseriti all’interno dell’header
+SOAP come campi custom. Erogatore e fruitore DEVONO inoltre seguire le
+seguenti regole:
+
+-  Le specifica delle interfacce del fruitore e dell’erogatore DEVONO
+   dichiarare tutti i metodi esposti con relativi schemi dei messaggi di
+   richiesta e di ritorno. Inoltre le interfacce devono specificare
+   eventuali header SOAP richiesti;
+
+-  La specifica dell’interfaccia del fruitore DEVE rispettare quanto
+   richiesto dall’erogatore; in particolare, non esistendo un
+   equivalente in WSDL del formalismo per le callback, si richiede che
+   l’erogatore fornisca un WSDL di esempio su un endpoint differente da
+   quello del servizio; il fruitore svilupperà quindi un servizio che
+   rispetta questa specifica al fine di fornire un endpoint di callback;
+
+-  Al passo (1), il fruitore DEVE indicare l’endpoint della callback
+   utilizzando l’header SOAP custom X-ReplyTo;
+
+-  Al passo (2), l’erogatore DEVE fornire insieme all’acknowledgement
+   della richiesta nel body, il correlation ID utilizzando l’header SOAP
+   custom X-CorrelationID;
+
+-  Al passo (3), l’erogatore DEVE riutilizzare lo stesso correlation ID
+   fornito al passo (2) sempre utilizzando l’header SOAP custom
+   X-CorrelationID;
+
+-  Al passo (4), il fruitore DEVE riconoscere tramite un messaggio di
+   acknowledgement il ricevimento della risposta.
+
+Il corpo dei messaggi HTTP scambiati durante l’interazione DEVE seguire
+lo standard XML.
+
+.. _regole-di-processamento-3:
+
+Regole di processamento
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Sebbene la specifica SOAP proponga l’utilizzo dei codici di stato HTTP
```

Comment:
afaik SOAP always uses 500 and, in some well specific cases 4xx. See http://www.ws-i.org/Profiles/BasicProfile-1.0-2004-04-16.html#refinement16488480